### PR TITLE
Fix a crash occurring when first year not played

### PR DESCRIPTION
**Pull request by guilpier-code**
Sometimes, a unit test from **simple-study.cpp** won't pass. 
This test is named **two_MC_years__thermal_cluster_fullfills_area_demand_on_2nd_year_as_well**.

**Cause** : 
when executing code (see **solver.hxx**) : 
```c++
auto task = std::make_shared<yearJob<ImplementationType>>(
        ...
        y,
        ...
        numSpace,
        ...
        state[numSpace],
        ...);
```
When a year **y** is inactive in the playlist, variable **numSpace** is 99999, and `state[numSpace]` is undefined.

**More thoughts** : 
- this crash was first introduced in PR #2128 
- Associated commit : **be3301cdb3b2b7ed232db72e68a874c353d51609**
- It's unclear why we did not catch this crash sooner. Test **two_MC_years__thermal_cluster_fullfills_area_demand_on_2nd_year_as_well** should have failed already.
  Anyway, it fails at every run on **Windows** and in **debug** mode

### CODE REVIEW

**Comment by flomnes** [comment]
@guilpier-code Good catch ! Can you point an action run that failed ? I'm surprised we didn't catch this bug earlier.

Also, could you handle the formatting issue ?

**Comment by guilpier-code** [comment]
> @guilpier-code Good catch ! Can you point an action run that failed ? I'm surprised we didn't catch this bug earlier.

Please see the current PR's presentation comment. I updated it to answer your question.

> Also, could you handle the formatting issue ?

Done

File: src/solver/simulation/include/antares/solver/simulation/solver.hxx [hunk 1]
```diff
@@ -65,14 +65,14 @@ public:
...
-        state(pState),
+        states(pStates),
         yearByYear(pYearByYear),
         pDurationCollector(durationCollector),
         pResultWriter(resultWriter),
         simulationObserver_(simulationObserver),
         hydroManagement(study.areas, study.parameters, study.calendar, resultWriter)
     {
-        scratchmap = study.areas.buildScratchMap(numSpace);
+        // scratchmap = study.areas.buildScratchMap(numSpace);
```
Comment (flomnes) [inline src/solver/simulation/include/antares/solver/simulation/solver.hxx]:
```suggestion
```

File: src/solver/simulation/include/antares/solver/simulation/solver.hxx [hunk 2]
```diff
@@ -89,13 +89,13 @@ private:
...
     Data::Study& study;
-    Variable::State& state;
+    std::vector<Variable::State>& states;
     bool yearByYear;
     Benchmarking::DurationCollector& pDurationCollector;
     IResultWriter& pResultWriter;
     std::reference_wrapper<ISimulationObserver> simulationObserver_;
     HydroManagement hydroManagement;
-    Antares::Data::Area::ScratchMap scratchmap;
+    // Antares::Data::Area::ScratchMap scratchmap;
```
Comment (flomnes) [inline src/solver/simulation/include/antares/solver/simulation/solver.hxx]:
```suggestion
```

**Review by flomnes — COMMENTED** [review]
Remove commented-out code

File: src/solver/simulation/include/antares/solver/simulation/solver.hxx (hunk 1 above)
Comment (guilpier-code) [reply src/solver/simulation/include/antares/solver/simulation/solver.hxx]: Done

File: src/solver/simulation/include/antares/solver/simulation/solver.hxx (hunk 2 above)
Comment (guilpier-code) [reply src/solver/simulation/include/antares/solver/simulation/solver.hxx]: Done

**Review by flomnes — APPROVED** [review]

**Pull request merged**